analysis/study_definition.py

```diff
@@ -570,29 +572,125 @@
             "incidence": 0.95,
         },
     ),
+    # Fetch the comparator (<, >=, = etc) associated with a numeric value.
+    # Where a lab result is returned as e.g. <9.5 the numeric_value component
+    # will contain only the value 9.5 and you will need to use this function
+    # to fetch the comparator into a separate column.
+    # https://docs.opensafely.org/study-def-variables/#cohortextractor.patients.comparator_from
+    egfr_comparator=patients.comparator_from(
+        "egfr",
+        return_expectations={
+            "rate": "universal",
+            "category": {
+                "ratios": {  # ~, =, >= , > , < , <=
+                    None: 0.10,
+                    "~": 0.05,
+                    "=": 0.65,
+                    ">=": 0.05,
+                    ">": 0.05,
+                    "<": 0.05,
+                    "<=": 0.05,
+                }
+            },
+            "incidence": 0.80,
+        },
+    ),
+    egfr_category=patients.categorised_as(
+        {
+            "0": "DEFAULT",
+            "1": """
+                egfr_flag AND
+                    (egfr>=60 AND NOT
+                        (egfr_comparator = '<' AND NOT
+                        egfr_comparator = '<=' AND NOT
+                        egfr_comparator = '~'))
```

Comment:
Should these instead use `OR`? I don't think this would currently exclude those with `<=` or `~`

Reply:
Oh, good catch. This wouldn't work indeed! Changed to OR.